### ISSUE TITLE
Confusing error from `metaflow.S3` if Metaflow is not configured properly

### ISSUE THREAD

**Issue by tuulos** [Netflix/metaflow#683]
If Metaflow is not configured properly (e.g. no config file at all), The `S3DemoFlow` at  https://docs.metaflow.org/metaflow/data#store-and-load-objects-in-a-metaflow-flow fails with a confusing error. We could show a friendlier error

```2021-09-07 01:25:02.077 Workflow starting (run-id 1630977902074982):
2021-09-07 01:25:02.089 [1630977902074982/start/1 (pid 31904)] Task is starting.
2021-09-07 01:25:02.339 [1630977902074982/start/1 (pid 31904)] <flow S3DemoFlow step start> failed:
2021-09-07 01:25:02.343 [1630977902074982/start/1 (pid 31904)] Internal error
2021-09-07 01:25:02.344 [1630977902074982/start/1 (pid 31904)] Traceback (most recent call last):
2021-09-07 01:25:02.344 [1630977902074982/start/1 (pid 31904)] File "/home/ec2-user/metaflow/metaflow/cli.py", line 1007, in main
2021-09-07 01:25:02.344 [1630977902074982/start/1 (pid 31904)] start(auto_envvar_prefix='METAFLOW', obj=state)
2021-09-07 01:25:02.344 [1630977902074982/start/1 (pid 31904)] File "/home/ec2-user/.local/lib/python3.7/site-packages/click/core.py", line 1137, in __call__
2021-09-07 01:25:02.344 [1630977902074982/start/1 (pid 31904)] return self.main(args, kwargs)
2021-09-07 01:25:02.345 [1630977902074982/start/1 (pid 31904)] File "/home/ec2-user/.local/lib/python3.7/site-packages/click/core.py", line 1062, in main
2021-09-07 01:25:02.345 [1630977902074982/start/1 (pid 31904)] rv = self.invoke(ctx)
2021-09-07 01:25:02.345 [1630977902074982/start/1 (pid 31904)] File "/home/ec2-user/.local/lib/python3.7/site-packages/click/core.py", line 1668, in invoke
2021-09-07 01:25:02.345 [1630977902074982/start/1 (pid 31904)] return _process_result(sub_ctx.command.invoke(sub_ctx))
2021-09-07 01:25:02.345 [1630977902074982/start/1 (pid 31904)] File "/home/ec2-user/.local/lib/python3.7/site-packages/click/core.py", line 1404, in invoke
2021-09-07 01:25:02.345 [1630977902074982/start/1 (pid 31904)] return ctx.invoke(self.callback, ctx.params)
2021-09-07 01:25:02.345 [1630977902074982/start/1 (pid 31904)] File "/home/ec2-user/.local/lib/python3.7/site-packages/click/core.py", line 763, in invoke
2021-09-07 01:25:02.345 [1630977902074982/start/1 (pid 31904)] return __callback(args, kwargs)
2021-09-07 01:25:02.345 [1630977902074982/start/1 (pid 31904)] File "/home/ec2-user/.local/lib/python3.7/site-packages/click/decorators.py", line 26, in new_func
2021-09-07 01:25:02.345 [1630977902074982/start/1 (pid 31904)] return f(get_current_context(), args, kwargs)
2021-09-07 01:25:02.388 [1630977902074982/start/1 (pid 31904)] File "/home/ec2-user/metaflow/metaflow/cli.py", line 522, in step
2021-09-07 01:25:02.388 [1630977902074982/start/1 (pid 31904)] max_user_code_retries)
2021-09-07 01:25:02.388 [1630977902074982/start/1 (pid 31904)] File "/home/ec2-user/metaflow/metaflow/task.py", line 444, in run_step
2021-09-07 01:25:02.388 [1630977902074982/start/1 (pid 31904)] self._exec_step_function(step_func)
2021-09-07 01:25:02.388 [1630977902074982/start/1 (pid 31904)] File "/home/ec2-user/metaflow/metaflow/task.py", line 51, in _exec_step_function
2021-09-07 01:25:02.388 [1630977902074982/start/1 (pid 31904)] step_function()
2021-09-07 01:25:02.388 [1630977902074982/start/1 (pid 31904)] File "s3demo.py", line 8, in start
2021-09-07 01:25:02.388 [1630977902074982/start/1 (pid 31904)] with S3(run=self) as s3:
2021-09-07 01:25:02.388 [1630977902074982/start/1 (pid 31904)] File "/home/ec2-user/metaflow/metaflow/datatools/s3.py", line 287, in __init__
2021-09-07 01:25:02.388 [1630977902074982/start/1 (pid 31904)] current.run_id)
2021-09-07 01:25:02.388 [1630977902074982/start/1 (pid 31904)] File "/usr/lib64/python3.7/posixpath.py", line 94, in join
2021-09-07 01:25:02.389 [1630977902074982/start/1 (pid 31904)] genericpath._check_arg_types('join', a, p)
2021-09-07 01:25:02.389 [1630977902074982/start/1 (pid 31904)] File "/usr/lib64/python3.7/genericpath.py", line 155, in _check_arg_types
2021-09-07 01:25:02.389 [1630977902074982/start/1 (pid 31904)] raise TypeError("Can't mix strings and bytes in path components") from None
2021-09-07 01:25:02.389 [1630977902074982/start/1 (pid 31904)] TypeError: Can't mix strings and bytes in path components```

**Comment by himanshu007-creator** [comment]
Hi, I would like to work on this issue. Can you guide me further on what needs to be done?🙂

**Comment by saikonen** [comment]
should be fixed in release 2.9.13
